pokemonred_puffer/train.py

```diff
@@ -117,9 +117,9 @@ def agent_creator(env: gym.Env, args: pufferlib.namespace):
         else:
             policy = pufferlib.frameworks.cleanrl.Policy(policy)
 
+        policy = policy.to(args.train.device, non_blocking=True)
         if args.train.device == "cuda" and args.train.compile:
             torch.set_float32_matmul_precision(args.train.float32_matmul_precision)
```

Comment:
```suggestion
        if args.train.device == "cuda":
            torch.set_float32_matmul_precision(args.train.float32_matmul_precision)
            policy = policy.to(args.train.device, non_blocking=True)
            if args.train.compile:
              # indent compile related code here
```

I know `mps` and `tpu/xla` devices exist. But lets assume the only alternative device for training is CUDA. Generally, I haven't found a reason to have compile on without CUDA so I think it makes sense to place it under the if statement.

Reply:
Changed as suggested.